### PR TITLE
fix(dataset edit tools): clarifying `root` argument usage + adding related features

### PR DESCRIPTION
**Pull request by CarolinePascal**
## Type / Scope

- **Type**: Chore/Feature
- **Scope**: LeRobot Dataset editing tools

## Summary / Motivation

Following #3035, this PR clarifies the meaning of `root` in `lerobot-edit-dataset` :
- `root` describes the complete path to the input dataset, except for merge operations, where it defines the complete path to the output dataset
- `new_root` describes the complete path to the output dataset, except for split operations, where if defines the common path of all split datasets
- [NEW] `roots` is introduced in merge operations to describe a list of complete path to datasets to be merged

## Related issues

- Fixes / Closes: https://github.com/huggingface/lerobot/pull/3030, https://github.com/huggingface/lerobot/pull/2739, https://github.com/huggingface/lerobot/pull/1057, https://github.com/huggingface/lerobot/issues/2316
- Related: #3035 

## How was this tested (or how to run locally)

`pytest tests/datasets/test_dataset_tools.py` is green

Example: Check in `lerobot_edit_dataset.py`

## Checklist (required before merge)

- [ ] Linting/formatting run (`pre-commit run -a`)
- [ ] All tests pass locally (`pytest`)
- [ ] Documentation updated
- [ ] CI is green

## Reviewer notes

- Anything the reviewer should focus on (performance, edge-cases, specific files) or general notes.
- Anyone in the community is free to review the PR.

### CODE REVIEW

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 1]
```diff
@@ -60,11 +68,18 @@
         --operation.type merge \
         --operation.repo_ids "['lerobot/pusht_train', 'lerobot/pusht_val']"
 
+Merge multiple datasets to a specific output path:
+    lerobot-edit-dataset \
+        --repo_id lerobot/pusht_merged \
+        --root /path/to/pusht_merged \
+        --operation.type merge \
+        --operation.repo_ids "['lerobot/pusht_train', 'lerobot/pusht_val']"
+
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py:87]: fixed in e6e448b4759d4ea15233fbdb4c550510fdc558cc

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 2]
```diff
@@ -399,31 +453,24 @@ def handle_convert_image_to_video(cfg: EditDatasetConfig) -> None:
...
         output_repo_id = output_dir.name
         logging.info(f"Saving to local directory: {output_dir}")
     else:
-        # Auto-generate name: append "_video" to original repo_id
         output_repo_id = f"{cfg.repo_id}_video"
-        # Place new dataset as a sibling to the original dataset
-        # Extract just the dataset name (after last slash) for the local directory
-        local_dir_name = output_repo_id.split("/")[-1]
-        output_dir = dataset.root.parent / local_dir_name
+        output_dir = dataset.root.parent
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Fixed in 7f25b7d1e468cd444773a13bc2feba636be463bf

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 3]
```diff
@@ -399,31 +453,24 @@ def handle_convert_image_to_video(cfg: EditDatasetConfig) -> None:
...
+        output_repo_id = cfg.new_repo_id or f"{cfg.repo_id}_video"
+        logging.info(f"Saving to new_root: {output_dir}")
+    elif cfg.new_repo_id:
         output_repo_id = cfg.new_repo_id
-        # Place new dataset as a sibling to the original dataset
-        # Get the parent of the actual dataset root (not cfg.root which might be the lerobot cache dir)
-        # Extract just the dataset name (after last slash) for the local directory
-        local_dir_name = cfg.new_repo_id.split("/")[-1]
-        output_dir = dataset.root.parent / local_dir_name
+        output_dir = dataset.root.parent
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Fixed in 7f25b7d1e468cd444773a13bc2feba636be463bf

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 4]
```diff
@@ -292,16 +327,26 @@ def handle_merge(cfg: EditDatasetConfig) -> None:
...
+            return
 
-    logging.info(f"Loading {len(cfg.operation.repo_ids)} datasets to merge")
-    datasets = [LeRobotDataset(repo_id, root=cfg.root) for repo_id in cfg.operation.repo_ids]
+    if cfg.operation.repo_ids:
+        logging.info(f"Loading {len(cfg.operation.repo_ids)} datasets to merge")
+        datasets = [LeRobotDataset(repo_id) for repo_id in cfg.operation.repo_ids]
+    elif cfg.operation.roots:
+        logging.info(f"Loading {len(cfg.operation.roots)} datasets to merge")
+        datasets = [LeRobotDataset(repo_id="", root=root) for root in cfg.operation.roots]
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Fixed in a9f5b0b2740b7b01e0a9c50c7f35fdae63806cc8

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 5]
```diff
@@ -275,7 +305,12 @@ def handle_split(cfg: EditDatasetConfig) -> None:
     dataset = LeRobotDataset(cfg.repo_id, root=cfg.root)
 
     logging.info(f"Splitting dataset {cfg.repo_id} with splits: {cfg.operation.splits}")
-    split_datasets = split_dataset(dataset, splits=cfg.operation.splits)
+    split_datasets = split_dataset(
+        dataset,
+        splits=cfg.operation.splits,
+        output_dir=Path(cfg.new_root) if cfg.new_root else Path(cfg.root).parent if cfg.root else None,
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Fixed in 7f25b7d1e468cd444773a13bc2feba636be463bf

File: src/lerobot/datasets/dataset_tools.py [hunk 6]
```diff
@@ -103,7 +103,7 @@ def delete_episodes(
     logging.info(f"Deleting {len(episode_indices)} episodes from dataset")
 
     if repo_id is None:
-        repo_id = f"{dataset.repo_id}_modified"
+        repo_id = dataset.repo_id
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py]: Fixed in 82fa563553cfad140bc8ebc29dd9060188447f88

File: src/lerobot/datasets/dataset_tools.py [hunk 7]
```diff
@@ -330,7 +334,7 @@ def modify_features(
             raise ValueError(f"Cannot remove required features: {required_features}")
 
     if repo_id is None:
-        repo_id = f"{dataset.repo_id}_modified"
+        repo_id = dataset.repo_id
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py]: Fixed in 82fa563553cfad140bc8ebc29dd9060188447f88

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 8]
```diff
@@ -275,7 +305,12 @@ def handle_split(cfg: EditDatasetConfig) -> None:
...
+        repo_id=cfg.new_repo_id,
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Fixed in 82fa563553cfad140bc8ebc29dd9060188447f88

File: src/lerobot/datasets/dataset_tools.py [hunk 9]
```diff
@@ -1600,6 +1604,9 @@ def convert_image_to_video_dataset(
             # Video info will be updated after episodes are encoded
 
     # Create new metadata for video dataset
+    if repo_id is None:
+        repo_id = dataset.repo_id
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py]: Fixed in 82fa563553cfad140bc8ebc29dd9060188447f88

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 10]
```diff
@@ -275,33 +326,50 @@ def handle_split(cfg: EditDatasetConfig) -> None:
...
+
+    if cfg.operation.roots:
+        logging.info(f"Loading {len(cfg.operation.roots)} datasets to merge")
+        datasets = [
+            LeRobotDataset(repo_id=id, root=root)
+            for id, root in zip(cfg.operation.repo_ids, cfg.operation.roots, strict=True)
+        ]
+    else:
+        logging.info(f"Loading {len(cfg.operation.repo_ids)} datasets to merge")
+        datasets = [LeRobotDataset(repo_id) for repo_id in cfg.operation.repo_ids]
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py:368]: Fixed in 61f415842bff2f2b90f50c1da251f88b8196410b

File: src/lerobot/datasets/dataset_tools.py [hunk 11]
```diff
@@ -1548,8 +1552,8 @@ def convert_image_to_video_dataset(
 
     Args:
         dataset: The source LeRobot dataset with images
-        output_dir: Directory to save the new video dataset
-        repo_id: Repository ID for the new dataset (default: original_id + "_video")
+        output_dir: Root directory where the edited dataset will be stored. If not specified, defaults to $HF_LEROBOT_HOME/repo_id. Equivalent to new_root in EditDatasetConfig.
+        repo_id: Edited dataset identifier. When both output_dir and repo_id are not specified, modifications are applied in-place and a backup of the original dataset is created. Equivalent to new_repo_id in EditDatasetConfig.
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py]: Fixed in 61f415842bff2f2b90f50c1da251f88b8196410b

File: src/lerobot/datasets/dataset_tools.py [hunk 12]
```diff
@@ -390,8 +394,8 @@ def add_features(
     Args:
         dataset: The source LeRobotDataset.
         features: Dictionary mapping feature names to (feature_values, feature_info) tuples.
-        output_dir: Directory to save the new dataset. If None, uses default location.
-        repo_id: Repository ID for the new dataset. If None, appends "_modified" to original.
+        output_dir: Root directory where the edited dataset will be stored. If not specified, defaults to $HF_LEROBOT_HOME/repo_id. Equivalent to new_root in EditDatasetConfig.
+        repo_id: Edited dataset identifier. When both output_dir and repo_id are not specified, modifications are applied in-place and a backup of the original dataset is created. Equivalent to new_repo_id in EditDatasetConfig.
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py]: Fixed in 61f415842bff2f2b90f50c1da251f88b8196410b

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 13]
```diff
@@ -361,9 +433,14 @@ def handle_modify_tasks(cfg: EditDatasetConfig) -> None:
...
         raise ValueError("Must specify at least one of new_task or episode_tasks for modify_tasks operation")
 
-    # Warn about in-place modification behavior
-    if cfg.new_repo_id is not None:
-        logging.warning("modify_tasks modifies datasets in-place. The --new_repo_id parameter is ignored.")
+    if cfg.new_repo_id is not None or cfg.new_root is not None:
+        logging.warning(
+            "modify_tasks modifies datasets in-place. The --new_repo_id and --new_root parameters are ignored."
+        )
+        answer = input("Continue anyway? [y/N] ").strip().lower()
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Fixed in 61f415842bff2f2b90f50c1da251f88b8196410b

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 14]
```diff
@@ -275,33 +326,50 @@ def handle_split(cfg: EditDatasetConfig) -> None:
...
+    if cfg.operation.roots and len(cfg.operation.roots) != len(cfg.operation.repo_ids):
+        raise ValueError("repo_ids and roots must have the same length for merge operation")
 
-    logging.info(f"Loading {len(cfg.operation.repo_ids)} datasets to merge")
-    datasets = [LeRobotDataset(repo_id, root=cfg.root) for repo_id in cfg.operation.repo_ids]
+    if cfg.new_repo_id is not None or cfg.new_root is not None:
+        logging.warning(
+            "The --new_repo_id and --new_root parameters are ignored for merge operation, --repo_id and --root will be used instead."
+        )
+        answer = input("Continue anyway? [y/N] ").strip().lower()
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Fixed in 61f415842bff2f2b90f50c1da251f88b8196410b

File: src/lerobot/datasets/dataset_tools.py [hunk 15]
```diff
@@ -427,8 +431,8 @@ def remove_feature(
     Args:
         dataset: The source LeRobotDataset.
         feature_names: Name(s) of features to remove. Can be a single string or list.
-        output_dir: Directory to save the new dataset. If None, uses default location.
-        repo_id: Repository ID for the new dataset. If None, appends "_modified" to original.
+        output_dir: Root directory where the edited dataset will be stored. If not specified, defaults to $HF_LEROBOT_HOME/repo_id. Equivalent to new_root in EditDatasetConfig.
+        repo_id: Edited dataset identifier. When both output_dir and repo_id are not specified, modifications are applied in-place and a backup of the original dataset is created. Equivalent to new_repo_id in EditDatasetConfig.
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py]: Fixed in 61f415842bff2f2b90f50c1da251f88b8196410b

File: src/lerobot/datasets/dataset_tools.py [hunk 16]
```diff
@@ -288,8 +292,8 @@ def modify_features(
         dataset: The source LeRobotDataset.
         add_features: Optional dict mapping feature names to (feature_values, feature_info) tuples.
         remove_features: Optional feature name(s) to remove. Can be a single string or list.
-        output_dir: Directory to save the new dataset. If None, uses default location.
-        repo_id: Repository ID for the new dataset. If None, appends "_modified" to original.
+        output_dir: Root directory where the edited dataset will be stored. If not specified, defaults to $HF_LEROBOT_HOME/repo_id. Equivalent to new_root in EditDatasetConfig.
+        repo_id: Edited dataset identifier. When both output_dir and repo_id are not specified, modifications are applied in-place and a backup of the original dataset is created. Equivalent to new_repo_id in EditDatasetConfig.
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py]: Fixed in 61f415842bff2f2b90f50c1da251f88b8196410b

File: src/lerobot/datasets/dataset_tools.py [hunk 17]
```diff
@@ -89,8 +89,8 @@ def delete_episodes(
     Args:
         dataset: The source LeRobotDataset.
         episode_indices: List of episode indices to delete.
-        output_dir: Directory to save the new dataset. If None, uses default location.
-        repo_id: Repository ID for the new dataset. If None, appends "_modified" to original.
+        output_dir: Root directory where the edited dataset will be stored. If not specified, defaults to $HF_LEROBOT_HOME/repo_id. Equivalent to new_root in EditDatasetConfig.
+        repo_id: Edited dataset identifier. When both output_dir and repo_id are not specified, modifications are applied in-place and a backup of the original dataset is created. Equivalent to new_repo_id in EditDatasetConfig.
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py]: Fixed in 61f415842bff2f2b90f50c1da251f88b8196410b

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 18]
```diff
@@ -275,33 +326,50 @@ def handle_split(cfg: EditDatasetConfig) -> None:
...
+        if answer != "y":
+            logging.info("Aborted.")
+            return
+
+    if cfg.operation.roots:
+        logging.info(f"Loading {len(cfg.operation.roots)} datasets to merge")
+        datasets = [
+            LeRobotDataset(repo_id=id, root=root)
+            for id, root in zip(cfg.operation.repo_ids, cfg.operation.roots, strict=True)
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Fixed in 61f415842bff2f2b90f50c1da251f88b8196410b

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 19]
```diff
@@ -200,36 +237,46 @@ class ConvertImageToVideoConfig(OperationConfig):
...
+    if output_path == input_path:
+        backup_path = input_path.with_name(input_path.name + "_old")
 
-    return output_repo_id, output_dir
+        if input_path.exists():
+            if backup_path.exists():
+                shutil.rmtree(backup_path)
+            shutil.move(input_path, backup_path)
+
+    return output_repo_id, output_path
```
[the remark this answers is not in the document]
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py:275]: Fixed in 61f415842bff2f2b90f50c1da251f88b8196410b

File: src/lerobot/datasets/dataset_tools.py [hunk 20]
```diff
@@ -152,7 +152,7 @@ def split_dataset(
         dataset: The source LeRobotDataset to split.
         splits: Either a dict mapping split names to episode indices, or a dict mapping
                 split names to fractions (must sum to <= 1.0).
-        output_dir: Base directory for output datasets. If None, uses default location.
+        output_dir: Root directory where the split datasets will be stored. If not specified, defaults to $HF_LEROBOT_HOME.
```
Comment (s1lent4gnt) [inline src/lerobot/datasets/dataset_tools.py]: I think this should be $HF_LEROBOT_HOME/repo_id instead.

Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py]: `output_dir` is used as :
```
      split_output_dir = (
          output_dir / split_name if output_dir is not None else HF_LEROBOT_HOME / split_repo_id
      )
```

So technically it's just `$HF_LEROBOT_HOME`

Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py]: But I see what you mean, I will update the docstring !

Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py]: Fixed in [8b7f68a](https://github.com/huggingface/lerobot/pull/3049/commits/8b7f68a20a2682d9b01af5b06fafb3cc59d7f593)

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 21]
```diff
@@ -60,11 +78,25 @@
...
+        --operation.type merge \
+        --operation.repo_ids "['pusht_train', 'pusht_val']" \
+        --operation.roots "['/path/to/pusht_train', '/path/to/pusht_val']"
+
 Remove camera feature:
     lerobot-edit-dataset \
         --repo_id lerobot/pusht \
         --operation.type remove_feature \
-        --operation.feature_names "['observation.images.top']"
+        --operation.feature_names "['observation.image']"
```
Comment (s1lent4gnt) [inline src/lerobot/scripts/lerobot_edit_dataset.py:99]: Maybe we should keep `observation.images.top`

Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py:99]: In `pusht` we only have a `observation.image` feature, so this example would fail with `.top`

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 22]
```diff
@@ -272,19 +319,31 @@ def handle_split(cfg: EditDatasetConfig) -> None:
...
         )
 
+    if cfg.new_repo_id is not None:
+        logging.warning(
+            "split uses the original dataset identifier --repo_id to generate split names. The --new_repo_id parameter is ignored."
+        )
+        answer = input("Continue anyway? [y/N] ").strip().lower()
+        if answer != "y":
+            logging.info("Aborted.")
+            return
```
Comment (s1lent4gnt) [inline src/lerobot/scripts/lerobot_edit_dataset.py]: input() will raise EOFError in non-interactive environments (e.g. CI pipeline)

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 23]
```diff
@@ -295,13 +354,28 @@ def handle_merge(cfg: EditDatasetConfig) -> None:
...
+        datasets = [
+            LeRobotDataset(repo_id=repo_id, root=root)
+            for repo_id, root in zip(cfg.operation.repo_ids, cfg.operation.roots, strict=True)
+        ]
+    else:
+        logging.info(f"Loading {len(cfg.operation.repo_ids)} datasets to merge")
+        datasets = [LeRobotDataset(repo_id) for repo_id in cfg.operation.repo_ids]
 
-    output_dir = Path(cfg.root) / cfg.repo_id if cfg.root else HF_LEROBOT_HOME / cfg.repo_id
+    output_dir = Path(cfg.root) if cfg.root else HF_LEROBOT_HOME / cfg.repo_id
```
Comment (s1lent4gnt) [inline src/lerobot/scripts/lerobot_edit_dataset.py]: I think it's better to keep --root as the input path (consistent with other handlers) and use --new_root for the merge output. That way --root means the same thing everywhere, and the warning/prompt for --new_root being ignored becomes unnecessary.

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 24]
```diff
@@ -329,11 +403,15 @@ def handle_remove_feature(cfg: EditDatasetConfig) -> None:
 
     dataset = LeRobotDataset(cfg.repo_id, root=cfg.root)
     output_repo_id, output_dir = get_output_path(
-        cfg.repo_id, cfg.new_repo_id, Path(cfg.root) if cfg.root else None
+        cfg.repo_id,
+        new_repo_id=cfg.new_repo_id,
+        root=cfg.root,
+        new_root=cfg.new_root,
```
Comment (s1lent4gnt) [inline src/lerobot/scripts/lerobot_edit_dataset.py:401]: Type hints are wrong. We can either change the function signature to accept `str` or cast to `Path()` in the function call.

Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py:401]: Function hints extended in b1e05a390f9e8674a3beaa08cb097c06518599af

File: src/lerobot/scripts/lerobot_edit_dataset.py (hunk 22 above)
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Yes, I saw Copilot complain about it. I checked we only used the `handle_` functions in a script that is supposed to be run by the user in the terminal, so I thought it was OK 🤔 

I can remove the y/N choice though as we always create a backup in case of in-place modification !

File: src/lerobot/datasets/dataset_tools.py [hunk 25]
```diff
@@ -1600,6 +1600,7 @@ def convert_image_to_video_dataset(
             # Video info will be updated after episodes are encoded
 
     # Create new metadata for video dataset
+    output_dir = Path(output_dir) if output_dir is not None else HF_LEROBOT_HOME / repo_id
```
Comment (s1lent4gnt) [inline src/lerobot/datasets/dataset_tools.py:1603]: `repo_id` and `output_dir` are both optional, but if both are `None` this line crashes. Either guard `repo_id` or keep `output_dir` as required (the edit script always provides it anyway).

File: src/lerobot/scripts/lerobot_edit_dataset.py (hunk 23 above)
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: I agree, but this would be a breaking change for people already using the tool. That being said, we can always add a warning, but this is simply shifting the warning from one variable to another.

File: src/lerobot/scripts/lerobot_edit_dataset.py (hunk 22 above)
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Done in [33767a0](https://github.com/huggingface/lerobot/pull/3049/commits/33767a0cdc949e2469a526aefa8662eb64835d2f)

File: src/lerobot/datasets/dataset_tools.py (hunk 25 above)
Comment (CarolinePascal) [reply src/lerobot/datasets/dataset_tools.py:1603]: It should be OK, as we check for `repo_id` before (l.1583) and assign it to a value if needed :
```
if repo_id is None:
    repo_id = f"{dataset.repo_id}_video"
```

File: src/lerobot/datasets/dataset_tools.py [hunk 26]
```diff
@@ -1529,7 +1529,7 @@ def modify_tasks(
 
 def convert_image_to_video_dataset(
     dataset: LeRobotDataset,
-    output_dir: Path,
+    output_dir: Path | None = None,
```
Comment (s1lent4gnt) [inline src/lerobot/datasets/dataset_tools.py:1532]: I'd prefer keeping output_dir required here. Path resolution should stay in the script layer, not the library layer for separation of concerns.

File: src/lerobot/scripts/lerobot_edit_dataset.py (hunk 23 above)
Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Done in 92b66e996cbe6c46c416d0704bb31f3506a8ed1b

File: src/lerobot/scripts/lerobot_edit_dataset.py [hunk 27]
```diff
@@ -200,36 +233,46 @@ class ConvertImageToVideoConfig(OperationConfig):
...
 
 
 @dataclass
 class EditDatasetConfig:
-    repo_id: str
+    # Operation configuration.
     operation: OperationConfig
+    # Input dataset identifier. Always required unless for Merge operation.
+    repo_id: str | None = None
+    # Root directory where the input dataset is stored. If not specified, defaults to $HF_LEROBOT_HOME/repo_id. For Merge operation, this is the output dataset directory.
```
Comment (s1lent4gnt) [inline src/lerobot/scripts/lerobot_edit_dataset.py]: This should be changed, as new_root is the output for merge operation.

Comment (CarolinePascal) [reply src/lerobot/scripts/lerobot_edit_dataset.py]: Solved in [e2c2c06](https://github.com/huggingface/lerobot/pull/3049/commits/e2c2c06141399d082e4adee5559071810ce3a13c)
